### DISCUSSION TITLE
Cheat sheet for camera shots, angles and syntax

### DISCUSSION THREAD

**Discussion by monkrus** [lllyasviel/Fooocus#1993] [Show and tell]
![Camera-Shots-Chart-of-Camera-Shot-Sizes-Diagram-Infographic-StudioBinder](https://github.com/lllyasviel/Fooocus/assets/12500447/e695ec57-735b-49f3-b5d1-39ece15a12c6)
![Camera-Shots-Level-and-Shot-Sizes-Diagram-and-Chart-Infographic-StudioBinder](https://github.com/lllyasviel/Fooocus/assets/12500447/91c6ae97-bade-4dc7-9ef3-0ba7ee88e4a0)
<img width="371" alt="image" src="https://github.com/lllyasviel/Fooocus/assets/12500447/7a0e28f3-1011-442d-8274-e0fd2cee8533">

**Comment by jatin-nandwani** [comment]
Thing is how do we apply this in the prompt, any examples pls

**Reply by poor7** [reply]
It’s not difficult to make a promts using this cheat sheet, but it won’t always work correctly. For example, for a "cowboy shot", you will often draw cowboys, etc.

**Reply by monkrus** [reply]
I agree, this is just a helper. Also, prompt order matters; the closer it is to the beginning, the more effect it has.

**Comment by Hypersonic-Popcorn** [comment]
I tend to use "a [shot type] photo of [subject] [details] [background]" as my formula. Like poor7 mentioned, many of these shot names either don't work or work unreliably. The model being used as well as any loras will affect what shots work. 
The styles can also have an effect so you may need to raise the prompt strength to get the shot you want. Also, the dimensions of the photo may change things i.e. if the photo is taller than it is wide, extreme close up shots might only show one eye.

**Comment by Hypersonic-Popcorn** [comment]
So I'm testing each of these to see how well they work with the OpenDalle 1.1 model with and without the default styles (Fooocus v2, Enhance, and Sharp). The prompt was just "a [shot name] photo of a woman"

I was going to go into detail for each shot type but it seems that the same things apply to most shots. Extreme close up works best in landscape modes, everything else works best with portrait. Using the default styles seems to bring the "camera" closer in with most shot types. So a full shot with styles usually results in a cowboy shot. Cowboy shot makes cowboys, no exceptions :) Anything with "medium" in it seems to produce a lot of black and white photos with the models I tested (OpenDalle 1.1, RunDiffusionV7, NightVisionXL)

The camera level prompts work better than I remembered, Styles were likely to reinterpret the shot (ground level usually put the subject lying on the ground) so additional prompting would be required.

**Comment by xhoxye** [comment]
`a masterpiece of (Waist Shot:1.3) of (a woman:1.1), (From England:1.1), (Young Adult:1.1), (Three-Quarter View:1.1), (OOTD:0.9), (by Contemporary Photographer:0.9), `
https://github.com/xhoxye/Fooocus-up_mask/blob/Fooocus_wildcards_enhance/wildcards/xhox_shot.txt
Here's my formula and PR.
![QQ截图20240121115513](https://github.com/lllyasviel/Fooocus/assets/129571231/2026b2fc-e165-4c33-8bc3-91eca5685776)

**Reply by metercai** [reply]
@xhoxye  可以有更好的组织方式，方便的话加qq群：938075852

**Reply by lgwjames** [reply]
can this be added to the one i installed?

**Reply by GodFazer** [reply]
how did u do this much loras?

**Comment by monkrus** [comment]
To create a full-height image, experiment with the following prompt example with 3:5 ratio : [full-height portrait], [object], [elements of clothing], e.g. "full height portrait, French girl, shorts, sneakers". Looks like added point help to create the desired image.

**Comment by monkrus** [comment]
You might also consider this approach:
**AI Image Generation Prompt Template**


Subject Description:
- Main Subject: [Describe the central figure or object]
- Key Features: [Color, size, distinctive characteristics]


Scene Setting:
- Environment: [Indoors, outdoors, specific location (e.g., forest, city, beach)]
- Atmosphere: [Time of day, weather, any environmental effects (e.g., foggy, rainy)]
- Contextual Elements: [Additional features or objects surrounding the subject]


Creative Direction:
- Art Style: [Choose from realistic, abstract, impressionistic, etc.]
- Mood: [Specify the emotional feel of the image, e.g., joyful, somber]
- Influences: [Mention artistic influences or styles to draw inspiration from]


Composition and Perspective:
- Viewpoint: [First-person POV, aerial view, side profile, etc.]
- Composition: [Arrangement of elements, focus points]


Abstract Qualities (if applicable):
- Concept/Idea: [Describe any abstract concept or theme represented in the image]


Language and Descriptors:
- Sensory Description: [Include sensory details: touch, sound, etc., to add depth]
- Specific Descriptors: [Adjectives and adverbs to refine appearance, texture, or action]



**Sample Prompt:**
**P.S. The default settings were used to generate those samples.**

Generate an [Resolution]-resolution image depicting [Main Subject] with [Key Features], set against [Environment] that embodies [Atmosphere]. The scene should include [Contextual Elements], with a focus on [Additional Features] that enhances the [Mood] of the image. Adopt an [Art Style] approach, taking inspiration from [Influences] for artistic direction. The composition should provide a [Viewpoint], showing [Composition]. Ensure that the image conveys [Concept/Idea] through its visual narrative. Incorporate elements from the [Reference Images] or aesthetics similar to [Artistic Reference].

**Prompt Example 1:**
Generate a high-resolution, digital painting of a serene, enchanted forest at twilight. In the foreground, a gentle, clear stream meanders through the scene, reflecting the soft, fading light. Ancient trees with broad, twisted trunks and luminous, bioluminescent leaves tower over a carpet of vibrant, multicolored wildflowers. A subtle mist hangs in the air, giving the forest an ethereal, magical quality. The art style should be semi-realistic, reminiscent of a Studio Ghibli animation. Emphasize a peaceful, whimsical mood, and use a perspective that places the viewer at eye level with the middle of the forest, as if they’ve just stumbled upon this hidden wonder.
![image](https://github.com/lllyasviel/Fooocus/assets/12500447/26e6f493-6099-40d3-bf1f-a877e58de248)

**Prompt example 2:**
Generate a high-resolution image portraying a young, attractive girl in her early twenties with radiant skin and shoulder-length curly hair under a city park’s cherry blossom tree. Ensure the lighting is warm, emulating a sunny afternoon glow, and aim for a joyful and carefree mood reminiscent of a high-end fashion editorial. The art style should be realistic yet slightly dreamy. The girl should be looking over her shoulder with a cheerful expression, engaging the viewer. Capture the feeling of youth and the joy of spring, depicting the softness of the atmosphere and the vibrant energy of life.



![image (1)](https://github.com/lllyasviel/Fooocus/assets/12500447/0cd9b109-c135-42c8-aeaa-9ae033eb8c8f)

**Reply by poor7** [reply]
Let's say you're right. Then tell me what kind of word soup I need to write to get the same girl but with a greater depth of field?

**Reply by monkrus** [reply]
Did you mean the more focused background?

**Reply by poor7** [reply]
Yes.

**Reply by monkrus** [reply]
Have you tried your trick with "(outdoors-focus:1.8)"?

**Reply by poor7** [reply]
Yes, I tried it. It partially works.

**Comment by monkrus** [comment]
I afraid I dont have a better answer for you.
